src/Socket/socket.ts

```diff
@@ -463,7 +463,7 @@ export const makeSocket = (config: SocketConfig) => {
 		const pairingCode = customPairingCode ?? bytesToCrockford(randomBytes(5))
 
 		if (customPairingCode && customPairingCode?.length !== 8) {
-			throw new Error('Custom pairing code must be exactly 8 chars')
+			throw new Boom('Custom pairing code must be 8 characters', { statusCode: 422 })
```

Comment:
```suggestion
			throw new Boom('Custom pairing code must be 8 characters', { statusCode: 400 })
```
422 is incorrect. 422 refers to a server side error in processing the request. 
400 is the right HTML code when your request or its data is incorrect.

I don't see the point in throwing an error code, personally. Might be useful for people handling this function call, but I think Booms end the socket iirc (so this PR is useless and even harmful), or people should handle this logic before calling Baileys. Users of Baileys need just as much validation on the user end.